aten/src/ATen/core/op_registration/hacky_wrapper_for_legacy_signatures.h

```diff
@@ -207,12 +208,111 @@ constexpr auto with_explicit_optional_tensors(KernelFunc kernel_func) {
     return kernel_func;
 }
 
+template<class Arg> constexpr bool is_out_argument_() {
+    return std::is_same<at::Tensor&, Arg>::value;
 }
+template<class Arg> using is_out_argument = guts::bool_constant<is_out_argument_<Arg>()>;
 
-template<class TargetSignature, class FuncPtr>
+template<class OutTensorArgIndices, class KernelFunc>
+struct with_out_arguments_reordered_impl final {
+private:
+    // For an example op
+    //   > aten::example(Tensor a, int64_t b, Tensor(a!) out_c, int64_t d, Tensor(b!) out_e) -> (Tensor(a!), Tensor(b!))
```

Comment:
This example op has out arguments sprinkled through the signature, which AFAICT never happens in `native_functions.yaml`. I think it'd be clearer to use an example of the dominant pattern -  out args at the end. (Among other things it would avoid the confusion of describing ops like this as having out arguments at the end, as on line 277.)

More importantly, are there any actual ops where out arguments aren't contiguous at the end? I think this logic (both here and in codegen, but especially here) could be made quite a bit simpler if it didn't have to handle that class of signatures.

Reply:
You're right that the current implementation is more general than it would have to be. Not sure if restricting it to the actually occuring scenarios would simplify the logic though. Looking through it, the implementation could take two numbers [OutTensorArgIndicesBegin, OutTensorArgIndicesEnd] to represent a range instead of taking a full list of argument indices. But I think that's it, inside the metaprogramming I'd then still have to build the full OutTensorArgIndices array, so actually the metaprogramming would then contain an additional step and be more complex. Open to suggestions though if you have simplification ideas.